buildpack/descriptor.go

```diff
@@ -59,31 +112,55 @@ func (bg Group) Append(group ...Group) Group {
 	return bg
 }
 
-// A GroupBuildpack represents a buildpack referenced in a buildpack.toml's [[order.group]].
-// It may be a regular buildpack, or a meta buildpack.
-type GroupBuildpack struct {
-	API      string `toml:"api,omitempty" json:"-"`
+// A GroupElement represents a buildpack referenced in a buildpack.toml's [[order.group]] OR
+// a buildpack or extension in order.toml OR a buildpack or extension in group.toml.
+type GroupElement struct {
```

Comment:
The use of a single struct to represent these 3 subtly different representations of a buildpack and/or extension seems sub-optimal, but refactoring it would introduce a lot more changes to this already large PR.